src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
I'm not sure what `sizeof(OtherInt) <= sizeof(Int)` is protecting against. Isn't (a%b)<=a?

While I don't think we need to protect against the size, what about when `OtherInt` is signed and `Int` is unsigned? Couldn't we potentially try to store a signed integer into an unsigned result? That may be hard to protect against, though, as literals will be signed, and `a%2` would not compile when a is unsigned.